src/main/java/org/hpccsystems/jdbcdriver/HPCCDatabaseMetaData.java

```diff
@@ -1918,40 +1919,41 @@ private boolean fetchHPCCFilesInfo(String filename)
         return isSuccess;
     }
 
-    private int parseHPCCQuery(HPCCQuerySet[] querysets)
+    private int parseHPCCQuery(HPCCQuerySetWrapper[] querysets)
     {
         int hpccQueryParsedCount = 0;
 
         try
         {
             for (int querysetindex = 0; querysetindex < querysets.length; querysetindex++)
             {
-                HPCCQuerySet hpccQuerySet = querysets[querysetindex];
+                HPCCQuerySetWrapper hpccQuerySet = querysets[querysetindex];
                 String querySetName = hpccQuerySet.getName();
-                PublishedQuery[] querySetQueries = hpccQuerySet.getQuerySetQueries();
+                List<PublishedQueryWrapper> querySetQueryList = hpccQuerySet.getQuerySetQueries().getQuerySetQuery();
+                PublishedQueryWrapper[] querySetQueries = (PublishedQueryWrapper[]) querySetQueryList.toArray();
```

Comment:
Our static analysis engine detects that your code is problematic. According to the [Correctness](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [BC: Impossible downcast of toArray() result (BC_IMPOSSIBLE_DOWNCAST_OF_TOARRAY)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bc-impossible-downcast-of-toarray-result-bc-impossible-downcast-of-toarray).

This code is casting the result of calling `toArray()` on a collection to a type more specific than `Object[]`, as in:
```java
String[] getAsArray(Collection<String> c) {
    return (String[]) c.toArray();
}
```
This will usually fail by throwing a **ClassCastException**. The `toArray()` of almost all collections return an `Object[]`. They can't really do anything else, since the Collection object has no reference to the declared generic type of the collection.

The correct way to do get an array of a specific type from a collection is to use `c.toArray(new String[]);` or `c.toArray(new String[c.size()]);` (the latter is slightly more efficient).

There is one common/known exception to this. The `toArray()` method of lists returned by `Arrays.asList(...)` will return a covariantly typed array. For example, `Arrays.asArray(new String[] { "a" }).toArray()` will return a `String []`.